codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/protocols/parse/CborParserGenerator.kt

```diff
@@ -496,7 +530,7 @@ class CborParserGenerator(
                     if (this@CborParserGenerator.returnSymbolToParse(target).isUnconstrained) {
                         rust("decoder.string()")
                     } else {
-                        rust("#T::from(u.as_ref())", symbolProvider.toSymbol(target))
+                        rust("decoder.string().map(|s| s.as_str().into())")
```

Comment:
This path would only be hit by the server if the symbol is constrained. I prefer the previous codegen since with `#T::from()` it's apparent the type you want to convert into.

Reply:
Love the long explanation above, thanks a ton! Addressed David's original suggestion in 801af00